internal/model/labels.go

```diff
@@ -59,6 +61,39 @@ func (l Labels) Clone() Labels {
 	return cp
 }
 
+// Equal returns true when the two maps of labels are equal.
```

Comment:
I wonder if this is premature generalisation too. If we were to copy labels to a sorted slice when constructing the aggregation key, then we could do a more efficient comparison (i.e. by stepping through the slices in order). Also, by having the Equal method here, I would expect the Global field to be compared too - it's not relevant for the aggregation as we only compare global labels.